symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php

```diff
@@ -0,0 +1,350 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Attendance\Api;
+
+use DateTime;
+use Exception;
+use OrangeHRM\Attendance\Api\Model\AttendanceRecordModel;
+use OrangeHRM\Attendance\Exception\AttendanceServiceException;
+use OrangeHRM\Attendance\Traits\Service\AttendanceServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Entity\AttendanceRecord;
+
+class EmployeeAttendanceRecordAPI extends Endpoint implements CrudEndpoint
+{
+    use AttendanceServiceTrait;
+    use AuthUserTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_DATE = 'date';
+    public const PARAMETER_TIME = 'time';
+    public const PARAMETER_TIMEZONE_OFFSET = 'timezoneOffset';
+    public const PARAMETER_NOTE = 'note';
+
+    private ?bool $canUserChangeCurrentTime = null;
+    private ?bool $canUserModifyAttendance = null;
+    private ?bool $canSupervisorModifyAttendance = null;
```

Comment:
shall we move this status to service class?

Reply:
moved